src/System.Runtime/tests/System/StringTests.netcoreapp.cs

```diff
@@ -41,6 +42,72 @@ public static void Ctor_CharSpan(char[] valueArray, int startIndex, int length,
             Assert.Equal(expected, new string(span));
         }
 
+        [Fact]
+        public static unsafe void Ctor_CharPtr_DoesNotAccessInvalidPage()
+        {
+            // Allocates a buffer of all 'x' followed by a null terminator,
+            // then attempts to create a string instance from this at various offsets.
+
+            const int MaxCharCount = 128;
+            using BoundedMemory<char> boundedMemory = BoundedMemory.Allocate<char>(MaxCharCount);
+            boundedMemory.Span.Fill('x');
+            boundedMemory.Span[MaxCharCount - 1] = '\0';
+            boundedMemory.MakeReadonly();
+
+            using MemoryHandle memoryHandle = boundedMemory.Memory.Pin();
+
+            for (int i = 0; i < MaxCharCount; i++)
+            {
+                string expectedString = new string('x', MaxCharCount - i - 1);
+                string actualString = new string((char*)memoryHandle.Pointer + i);
+                Assert.Equal(expectedString, actualString);
+            }
+        }
+
+        [Fact]
+        public static unsafe void Ctor_SBytePtr_DoesNotAccessInvalidPage()
+        {
+            // Allocates a buffer of all ' ' followed by a null terminator,
+            // then attempts to create a string instance from this at various offsets.
+            // We use U+0020 SPACE instead of any other character because it lives
+            // at offset 0x20 across every supported code page.
+
+            IntPtr pAnsiStr = IntPtr.Zero;
+            try
+            {
+                pAnsiStr = Marshal.StringToHGlobalAnsi(" ");
+                if (((char*)pAnsiStr)[0] != ' ' || ((char*)pAnsiStr)[1] != '\0')
+                {
+                    // Current code page doesn't encode U+0020 SPACE as a single byte 0x20.
+                    // We don't know how to handle this in our unit test so we'll
+                    // just bail now and pretend we succeeded.
```

Comment:
It took me a minute to realize that the whole purpose of this whole try/finally is this one check.  It kind of obscures the test.  Could you move this out into a helper?